rotkehlchen/chain/ethereum/modules/lido/decoder.py

```diff
@@ -0,0 +1,526 @@
+                asset=A_STETH,
+                amount=steth_minted_tokens,
+                to_event_subtype=action_to_event_subtype,
+                to_notes=action_to_notes,
+                to_counterparty=CPT_LIDO_ETH,
+                paired_event_data=(paired_event, True),
+                extra_data={'staked_eth': str(collateral_amount)},
+            ))
+        else:  # did not break/find anything
+            log.error(
+                'At lido steth submit decoding of tx '
+                'not found the decoded event of the ETH transfer',
+            )
+            return DEFAULT_DECODING_OUTPUT
+
+        return DecodingOutput(action_items=action_items, matched_counterparty=CPT_LIDO_ETH)
+
+    def _map_erc20_lido_token_transfer(
+            self,
+            decoded_events: list['EvmEvent'],
+            transaction: 'EvmTransaction',  # pylint: disable=unused-argument
+            all_logs: list['EvmTxReceiptLog'],  # pylint: disable=unused-argument
+    ) -> list['EvmEvent']:
+        """Post decoding function to map the ERC20 Lido tokens reception"""
+        """TODO: To be checked with Rotki team, I am adding post decoding because I see that ERC20
+        transfers are not available in decoded_events at the time of decode by address rules
+        from what I have seen in the code. I got this solution looking at compound rules"""
+        for event in decoded_events:
+            if (
+                (event.counterparty == CPT_LIDO_ETH and event.notes is not None and event.extra_data is not None) and  # noqa: E501
+                (
+                    (event.event_type == HistoryEventType.RECEIVE and event.event_subtype == HistoryEventSubType.RECEIVE_WRAPPED) or  # noqa: E501
+                    (event.event_type == HistoryEventType.SPEND and event.event_subtype == HistoryEventSubType.RETURN_WRAPPED)  # noqa: E501
+                )
+            ):
+                event.notes = event.notes.format(amount=event.balance.amount)  # set the amount  # noqa: E501
+                break
+        return decoded_events
+
+    def _decode_lido_transfer_shares_for_steth_wrap(
+            self, context: DecoderContext,
+    ) -> DecodingOutput:
+        """Decode steth wrapping in wsteth from TransferShares log event"""
+
+        user_address = hex_or_bytes_to_address(context.tx_log.topics[1])
+
+        #  we need to retrieve the transferred stETH amount deterministically.
+        #  expected log index for the transferred steth is -1
+        #  expected log index for the transferred wsteth is -3
+        expected_steth_transfer_index = context.tx_log.log_index - 1
+        expected_wsteth_transfer_index = context.tx_log.log_index - 3
+        paired_wsteth_ratio = None
+        sth_wrapped_amount = None
+        wsteth_received_amount = None
+
+        for tx_log in context.all_logs:
+            if (
+                tx_log.log_index == expected_steth_transfer_index and
```

Comment:
do not count on the log index. Please as I told you in another PR this does not help and is not deterministic. Just cause you saw it in some events does not make it a rule. Log index is non-deterministic and depends on other events in the same transaction. Remove this as a check.